app/models/configuration_script.rb

```diff
@@ -1,3 +1,10 @@
 class ConfigurationScript < ActiveRecord::Base
   belongs_to :configuration_manager, :class_name => 'ManageIQ::Providers::ConfigurationManager'
+
+  def run
+    configuration_manager.with_provider_object do |cs|
+      jt = cs.job_templates.find(manager_ref)
```

Comment:
Is this saying "get me the native object for this configuration script?  If so, then I'd expect a `with_provider_object` method right on configuration_script that returns the `jt`.  Then this method would just call

```ruby
with_provider_object do |jt|
  jt.launch('extra_vars' => variables)
end
```

On that note, where is `variables` coming from?